common/wuwebview/wuwebview.cpp

```diff
@@ -921,6 +922,19 @@ void WuWebView::addInputsFromXml(const char *xml)
     addInputsFromPTree(pt.get());
 }
 
+IConstWUQuery* WuWebView::getEmbeddedQuery()
+{
+    if (!loadDll())
+        return NULL;
+
+    StringBuffer dllXML;
+    if (!getEmbeddedWorkUnitXML(dll, dllXML))
+        return NULL;
+
+    Owned<ILocalWorkUnit> embeddedWU = createLocalWorkUnit(dllXML.str());
```

Comment:
Good point.
I think it is acceptable if this code is executed as part of returning the archive - but it will be too expensive to call in the main wuinfo code.  It probably needs a flag (as you suggest in the JIRA) which can be checked to see if the dll contains an archive.